### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: db/queries.py [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    start_period = datetime.date(2021, month=1, day=1)
+    end_period = datetime.date(2022, month=1, day=1)
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__gt=start_period,
+        last_activity__lt=end_period).values(
+        "customer__first_name")
```
Comment (olena-hul) [inline db/queries.py]: Move closing parantheses to a new line

File: db/queries.py [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.order_by(
+        "-sum_of_spent_money").values_list(
+        "customer__first_name",
+        "customer__last_name",
+        "sum_of_spent_money"
+    )[:5]
```
Comment (olena-hul) [inline db/queries.py]: Fix identation to be more readable

File: db/queries.py [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
-    pass
+    return Customer.objects.filter(
+        Q(first_name__startswith="I") | Q(last_name__contains="o"))
 
 
 def bonuses_less_then_spent_money() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        active_bonuses__lt=F("sum_of_spent_money")).values(
+        "customer__phone_number")
```
Comment (olena-hul) [inline db/queries.py]: Fix identation

**Review by olena-hul — DISMISSED** [review]

**Review by katevasilenko — APPROVED** [review]
Looks good to me

**Pull request closed without merging**